### PR TITLE
CLI pretty printer improvements.

### LINKED ISSUE #4108: Include Commit messages in List Commit
**What is the goal / desired outcome?**
It would be great to be able to see commit messages at a glance when listing all commits from a repo, rather than needing to inspect each individual commit.

**If there is a way to accomplish this today via workaround, what does that require?**
Yes, one could write a simple bash/python script to list all commits in a repo then inspect them all and print out the relationship of commits:messages, obviously that is somewhat tedious/obtuse for a solution.

**(Optional) What is your proposal for a feature to solve this?**
I propose to add an additional flag to the "list commit" command that will allow the user to tell the method to include messages in the returned list, so multiple commit messages can be more easily browsed.

### PR DESCRIPTION
**Pull request by jdoliner**
This PR makes a few changes that users have been asking for to our CLI printing. Here's an example of what it looks like:

```
➜ pachctl list repo
NAME    CREATED            SIZE (MASTER) DESCRIPTION                       
montage About a minute ago 1.283MiB      Output repo for pipeline montage. 
edges   About a minute ago 133.6KiB      Output repo for pipeline edges.   
images  About a minute ago 238.3KiB      A repo for images.                
➜ pachctl list commit images 
REPO   BRANCH COMMIT                           STARTED        SIZE     PROGRESS DESCRIPTION
images master 0f04136baae042c799109c05f98845a7 24 seconds ago 238.3KiB ▇▇▇▇▇▇▇▇ Commit some more images. 
images master 6212d23dbcf14c569835307a9d171863 44 seconds ago 57.27KiB ▇▇▇▇▇▇▇▇ Commit some images.      
➜ pachctl list pipeline
NAME    VERSION INPUT                CREATED            STATE / LAST JOB  DESCRIPTION                                                           
montage 1       (edges:/ ⨯ images:/) About a minute ago running / success A pipeline that makes a montage of edge detected and original images. 
edges   1       images:/*            About a minute ago running / success A pipeline that runs edge detection on images. 
```

Changes include:
- all objects that have description fields (repos, commits and pipelines) now print the descriptions in the list view.
- removed parent and duration fields from `list commit`, parent field made the results into a wall of random characters which was hard to read and generally not useful, since the parent commit is normally the one right before. Even when it's not, it's generally not that useful, since the parent of a commit can simply be referenced as `<commit-id>^` which is more understandable. Duration was removed because it's not that interesting. Removing these allowed us to make space for the description field.
- repos created by pipelines (output repos, cron input repos, and git input repos) now have descriptions attached to them automatically.

Fixes #4108.

### CODE REVIEW

**Comment by JoeyZwicker** [comment]
Not that I liked the "duration" field or disagree with getting rid of it, but now how would one easily see if commits are open or closed from `list commit`? 

That's a common piece of useful info that is very often used in debugging and understanding the state of a branch

**Comment by JoeyZwicker** [comment]
One idea might be that instead of `started` and `duration` what if the header was `finished` with the field as a timestamp. That would still generally give the info of when a commit was finished which lets you track general timing information about when thing should be triggering and the field would be `-` when the commit is open.

**Comment by JoeyZwicker** [comment]
Note: @svekars added the "needs docs" tag to this PR because we should go through our existing examples and in each pipeline json spec, add a description field so it shows up nicely in list pipeline. Not an urgent thing to do nor does it block this PR.

**Comment by svekars** [comment]
@JoeyZwicker ack.

**Comment by jdoliner** [comment]
@JoeyZwicker you can actually still tell if a commit is finished because the size field will be a `-` instead of a number, but I think I like the idea of showing `Finished` instead of `Started` anyways.

File: src/server/pfs/pretty/pretty.go [hunk 1]
```diff
@@ -114,29 +115,26 @@ func PrintCommitInfo(w io.Writer, commitInfo *pfs.CommitInfo, fullTimestamps boo
...
+			int(commitInfo.SubvenantCommitsSuccess),
+			int(commitInfo.SubvenantCommitsTotal-commitInfo.SubvenantCommitsSuccess-commitInfo.SubvenantCommitsFailure),
+			int(commitInfo.SubvenantCommitsFailure)))
 	}
-	fmt.Fprintf(w, "%s\t\n", pretty.ProgressBar(
-		8,
-		int(commitInfo.SubvenantCommitsSuccess),
-		int(commitInfo.SubvenantCommitsTotal-commitInfo.SubvenantCommitsSuccess-commitInfo.SubvenantCommitsFailure),
-		int(commitInfo.SubvenantCommitsFailure)))
+	fmt.Fprintf(w, "%s\t\n", commitInfo.Description)
```
[the remark this answers is not in the document]
Comment (jdoliner) [reply src/server/pfs/pretty/pretty.go]: All pretty printers now end in a `fmt.Fprintln` except for those that are just a single line, in which case I include the `\n` in the lone print statement.

**Pull request merged**